File: Recursion/permutation.js
const getPermutations = (array) => {
    const permutations = []
      permutationsHelper(array, [], permutations)
      return permutations
  }
  
  const permutationsHelper = (array, current, permutations) => {
      if (!array.length && current.length) {
          permutations.push(current)
      } else {
          for (let i = 0; i < array.length; i++) {
              const newArray = array.slice(0, i).concat(array.slice(i + 1))
              const newPermutation = current.concat([array[i]])
              permutationsHelper(newArray, newPermutation, permutations)
          }
      }
  }
  

File: Recursion/permutation.py
def get_permutation(array):
    permutations = []
    permutations_helper(array, [], permutations)
    return permutations

def permutations_helper(array, current, permutations):
    if not len(array) and len(current):
        permutations.append(current)
    else:
        for i in range(len(array)):
            newArray = array[:i] + array[i+1:]
            newPermutation = current + [array[i]]
            permutations_helper(newArray, newPermutation, permutations)